File: src/Server/base/AlimentoInactivo.js
const express = require('express'); 
const multer = require('multer');
const fs = require('fs');
const path = require('path');
const db = require('./connection');
const router = express.Router();

// Eliminar un alimento
router.put("/:id", (req, res) => {  // Se asume que el id se pasa como parámetro en la URL
  const { id } = req.params;  // Obtener el id de los parámetros de la URL
  console.log("id recibido:", id);

  const query1 = `UPDATE stock_detalle SET Activo = 0 WHERE Id_Alimento = ?`;

  db.query(query1, [id], (err, result1) => {  // Usamos un arreglo para pasar el parámetro correctamente
      if (err) {
        console.log(err);  // Registra el error en consola
        return res.status(500).send("Error al eliminar el alimento");
      }
      res.json({ id, message: "Alimento eliminado con éxito" });
      console.log("Alimento inactivo");
  });
});

router.get("/", async (req,res) =>{
  const [existencia] = await queryAsync(
    `SELECT Id_Alimento, Es_perecedero FROM cat_alimento 
    WHERE Id_Usuario_Alta = ? AND Alimento LIKE %?% AND Activo = 1`,
    [Id_Usuario_Alta, nombre]
  );
});


module.exports = router;

File: src/serverP/base/AlimentoInactivo.py
from fastapi import APIRouter, HTTPException, Depends
from pydantic import BaseModel
from typing import Dict
from .connection import get_db_connection
import logging

alimento_inactivo_bp = APIRouter()

class AlimentoEliminarResponse(BaseModel):
    id: int
    message: str

@alimento_inactivo_bp.put("/{id}", response_model=AlimentoEliminarResponse)
async def eliminar_alimento(id: int, db=Depends(get_db_connection)):
    try:
        query1 = "UPDATE cat_alimento SET Activo = 0 WHERE Id_Alimento = ?"
        query2 = "UPDATE stock_detalle SET Activo = 0 WHERE Id_Alimento = ?"

        cursor = db.cursor()
        cursor.execute(query1, (id,))
        cursor.execute(query2, (id,))
        db.commit()
        cursor.close()

        logging.info("Alimento inactivo")
        return {"id": id, "message": "Alimento eliminado con éxito"}
    except Exception as e:
        logging.error(f"Error al eliminar el alimento: {str(e)}")
        raise HTTPException(status_code=500, detail="Error al eliminar el alimento")
